beacon-chain/blockchain/receive_block.go

```diff
@@ -143,6 +144,8 @@ func (s *Service) ReceiveBlockBatch(ctx context.Context, blocks []*ethpb.SignedB
 	}
 
 	if err := s.VerifyWeakSubjectivityRoot(s.ctx); err != nil {
+		// log.Fatalf will prevent defer from being called
+		cleanup()
```

Comment:
why not just call `span.End()` here instead of introducing a third variable?

We do this in other places as well (e.g. call span.End())

Reply:
https://github.com/prysmaticlabs/prysm/blob/master/beacon-chain/sync/pending_blocks_queue.go#L88

Reply:
I wanted to follow the same pattern everywhere, but I agree that `span.End` can just be called directly.

Reply:
Issue is this pattern is not used anywhere in the code base so this actually creates inconsistency. 
<img width="560" alt="Screen Shot 2020-10-04 at 7 40 01 AM" src="https://user-images.githubusercontent.com/21316537/95018464-d2c26580-0614-11eb-9f0e-4e3b58085b6a.png">

Here's a few options:

1.) Revert this PR to use `span.End` and we stick with `span.End`
2.) Revert this PR to use `span.End` and in the subsequent PR we replace all with `cleanup()`
3.) In this PR, we replace all `span.End` usages with `cleanup()`

I don't have a strong preference the option, you can choose one, but i do want some consistencies

Reply:
discussed offline and this was resolved